eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -155,11 +155,14 @@ object NodeParams {
     val maxAcceptedHtlcs = config.getInt("max-accepted-htlcs")
     require(maxAcceptedHtlcs <= Channel.MAX_ACCEPTED_HTLCS, s"max-accepted-htlcs must be lower than ${Channel.MAX_ACCEPTED_HTLCS}")
 
+    val publicAddresses = config.getStringList("server.public-ips").toList.map(ip => new InetSocketAddress(ip, config.getInt("server.port")))
+    require(publicAddresses.count(_.getAddress == null) == 0, s"server.public-ips must contain valid IP addresses or host names")
```

Comment:
This does check that the hostname can be resolved to an ip, but OTOH it makes it look like we support having hostnames as public-ip which we don't (see #471)

Reply:
In fact Eclair supports hostnames because java.net.InetSocketAddress does:
```
scala> new java.net.InetSocketAddress("github.com", 5555)
res0: java.net.InetSocketAddress = github.com/192.30.253.112:5555

scala> res0.getAddress
res1: java.net.InetAddress = github.com/192.30.253.112

scala> res1.getClass
res2: Class[_ <: java.net.InetAddress] = class java.net.Inet4Address
```
That means we need to validate public-ips with a regex or something.

Reply:
Indeed `InetSocketAddress` does resolve hostnames, but only *once*, and it leads users to think that they can enter their dyndns hostname and have it automagically refresh, which is not the case.